Library/Source/RuntimeImpl.h

```diff
@@ -71,7 +73,7 @@ namespace Babylon
         // occasionally need access to the env as well; m_env provides this
         // access when the env is available, reverting to nullptr once the env
         // is destroyed.
-        Babylon::Env* m_env{};
+        std::unique_ptr<Babylon::Env> m_env{};
```

Comment:
Having this thing not be directly bound to the thread on which it lives feels wrong. Does Apple need this? Is there a different way we could work around it?

Reply:
I don't think so. And there is more to come. Resize event from metal must be in sync with rendering. If I try to render to a smaller UI window, I get a crash. In the same trend, on Android, I get issues if the rendering thread is actively working while the activity/surfaceView is suspended.